target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

```diff
@@ -0,0 +1,22 @@
+#!/bin/ash
+
+[ "$ACTION" == "add" ] || exit 0
+
+PHYNBR=${DEVPATH##*/phy}
+
+[ -n $PHYNBR ] || exit 0
+
+. /lib/functions.sh
+. /lib/functions/system.sh
+
+board=$(board_name)
+
+case "$board" in
+	linksys,ea7500-v2)
+		hw_mac_addr=$(mtd_get_mac_ascii devinfo hw_mac_addr)
+		[ "$PHYNBR" = "0" ] &&  macaddr_add $hw_mac_addr 1  > /sys${DEVPATH}/macaddress
+		[ "$PHYNBR" = "1" ] &&  macaddr_add $hw_mac_addr 2  > /sys${DEVPATH}/macaddress
```

Comment:
Is this actually working?

Reply:
Yes it works correctly but only after the wifi is enabled. 
For example in the following snippet I enabled just the second interface and it gets the Belkin mac address (30:23:03:XX:XX:XX).
```
root@OpenWrt:~# iw dev
phy#1
	Interface wlan1
		ifindex 9
		wdev 0x100000001
		addr 00:0c:43:XX:XX:XX
		type managed
		txpower 6.00 dBm
		multicast TXQ:
			qsz-byt	qsz-pkt	flows	drops	marks	overlmt	hashcol	tx-bytes	tx-packets
			0	0	0	0	0	0	0	0		0
phy#0
	Interface wlan0
		ifindex 11
		wdev 0x2
		addr 30:23:03:XX:XX:XX
		ssid OpenWrt
		type AP
		channel 11 (2462 MHz), width: 20 MHz, center1: 2462 MHz
		txpower 20.00 dBm
		multicast TXQ:
			qsz-byt	qsz-pkt	flows	drops	marks	overlmt	hashcol	tx-bytes	tx-packets
			0	0	0	0	0	0	0	0		0
```
Without this script, the wifi interfaces get the Ralink mac address stored in the factory partition (00:0c:43:XX:XX:XX)